File: src/MediaItem.java
import java.io.Serializable;

/**
 * MediaItem class is used as a component in composite pattern.
 * 
 * @author      Juha Hirvasniemi <jhirvasn@paju.oulu.fi>, Tapio Korvala <korvatap@paju.oulu.fi>
 * @version     1.0
 * @since       2013-12-30
 */
public interface MediaItem extends Serializable {

	void print();
	void add(MediaItem item);
	String getTitle();
	String[] getRow();
	void deleteObjectById(int id);
	int getId();
}

File: web/MediaItem.php
<?php

/**
 * MediaItem class is used as a component in composite pattern.
 * Contains skeleton functions for other classes that inherit MediaItem.
 * 
 * @author      Juha Hirvasniemi <jhirvasn@paju.oulu.fi>, Tapio Korvala <korvatap@paju.oulu.fi>
 * @version     1.0
 * @since       2013-12-30
 */
interface MediaItem {

    public function info();
    public function add($item);
    public function id();
}
?>